WalletWasabi.Gui/MainWindow.xaml.cs

```diff
@@ -127,12 +127,6 @@ private async Task ClosingAsync()
 						}
 
 						Hide();
-						var wm = IoC.Get<IShell>().Documents?.OfType<WalletManagerViewModel>().FirstOrDefault();
```

Comment:
Why is this?

Reply:
We started the Hardware wallet enumeration loop in the vm. So when we want to stop that task I have to find the vm and stop the loop manually. 
That is ugly and this is one of the reasons to do this PR.